fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeStmtAnalyzer.java

```diff
@@ -102,69 +106,58 @@ public Void visitExecuteAsStatement(ExecuteAsStmt stmt, ConnectContext session)
-                        scramblePassword = MysqlPassword.makeScrambledPassword(stmt.getOriginalPassword());
-                    } else {
-                        scramblePassword = MysqlPassword.checkPassword(stmt.getOriginalPassword());
+        /**
+         * convert password to hashed password
+         */
+        private byte[] analysePassword(UserIdentity userIdent, String originalPassword, boolean isPasswordPlain,
+                                       boolean checkReuse) {
+            if (Strings.isNullOrEmpty(originalPassword)) {
+                return new byte[0];
+            }
+            try {
+                if (isPasswordPlain) {
+                    // plain password should check for validation & reuse
+                    Auth.validatePassword(originalPassword);
+                    if (checkReuse) {
+                        GlobalStateMgr.getCurrentState().getAuth().checkPasswordReuse(userIdent, originalPassword);
                     }
-                } catch (DdlException | AnalysisException e) {
-                    // TODO AnalysisException used to raise in all old methods is captured and translated to SemanticException
-                    // that is permitted to throw during analyzing phrase under the new framework for compatibility.
-                    // Remove it after all old methods migrate to the new framework
-                    throw new SemanticException(e.getMessage());
+                    // convert plain password to scramble
+                    return MysqlPassword.makeScrambledPassword(originalPassword);
+                } else {
+                    return MysqlPassword.checkPassword(originalPassword);
                 }
-            } else {
-                scramblePassword = new byte[0];
+            } catch (DdlException | AnalysisException e) {
+                // TODO AnalysisException used to raise in all old methods is captured and translated to SemanticException
+                // that is permitted to throw during analyzing phrase under the new framework for compatibility.
+                // Remove it after all old methods migrate to the new framework
+                throw new SemanticException(e.getMessage());
             }
-            stmt.setScramblePassword(scramblePassword);
+        }
 
+        public Void visitCreateAlterUserStmt(BaseCreateAlterUserStmt stmt, ConnectContext session) {
+            analyseUserAndCheckExist(stmt.getUserIdent(), session, stmt instanceof AlterUserStmt);
+            /*
+             * IDENTIFIED BY
+             */
+            stmt.setScramblePassword(
+                    analysePassword(stmt.getUserIdent(), stmt.getOriginalPassword(), stmt.isPasswordPlain(),
+                            stmt instanceof AlterUserStmt));
             /*
              * IDENTIFIED WITH
              */
             if (!Strings.isNullOrEmpty(stmt.getAuthPlugin())) {
+                stmt.setAuthPlugin(stmt.getAuthPlugin().toUpperCase());
```

Comment:
Auth plugin is already turned to uppercase in AstBuilder.

Reply:
done